src/components/Button.tsx

```diff
@@ -65,35 +74,74 @@ const gradientDuoToneClasses: Record<GradientDuoTone, string> = {
     'text-gray-900 bg-gradient-to-r from-red-200 via-red-300 to-yellow-200 hover:bg-gradient-to-bl focus:ring-4 focus:ring-red-100 dark:focus:ring-red-400',
 };
 
+const sizeClasses: Record<Size, string> = {
+  extraSmall: 'text-xs px-3 py-2',
+  small: 'text-sm px-3 py-2',
+  medium: 'text-sm px-5 py-2.5',
+  large: 'text-base px-5 py-3',
+  extraLarge: 'text-base px-6 py-3.5',
+};
+
 export const Button: FC<ButtonProps> = ({
   children,
   pill,
   outline,
+  disabled = false,
+  loader = false,
+  iconButton = false,
+  label,
+  size = 'medium',
+  icon: Icon,
+  iconPosition,
   color = 'blue',
   gradientMonochrome,
   gradientDuoTone,
   ...props
 }) => {
   return (
     <button
+      disabled={loader || disabled}
```

Comment:
We should not force the button to be disabled if it is loader, we should let the disabled decision to the developers.